src/Harald.WebApi/Features/Connections/Infrastructure/Persistence/FindConnectionsByClientTypeClientIdChannelTypeChannelIdHandler.cs

```diff
@@ -23,30 +23,27 @@ public class FindConnectionsByClientTypeClientIdChannelTypeChannelIdHandler : IQ
         public async Task<IEnumerable<Connection>> HandleAsync(
             FindConnectionsByClientTypeClientIdChannelTypeChannelId query)
         {
-         
-            
-            
-            if (query.ChannelType != null && query.ChannelType.GetType() != typeof(ChannelTypeSlack) )
+            if (query.ChannelType != null && !query.ChannelType.IsEmpty && !(query.ChannelType is ChannelTypeSlack))
             {
                 return Enumerable.Empty<Connection>();
             }
 
-            if (query.ClientType != null && query.ClientType.GetType() != typeof(ClientTypeCapability))
+            if (query.ClientType != null && !query.ClientType.IsEmpty && !(query.ClientType is ClientTypeCapability))
             {
                 return Enumerable.Empty<Connection>();
             }
 
             var capabilities = await _capabilityRepository.GetAll();
 
-            if (query.ChannelId != null)
+            if (query.ChannelId != null && !query.ChannelId.IsEmpty)
```

Comment:
We could drop the IsEmpty check if we used the ChannelId object and made sure you couldn't give it a empty/space string

Reply:
After I fixed the explicit operator our null checks didnt fail when the inner value was null, so your right we could do without it. I decided to leave it in as long as its possible to factorize SubstitutableString objects with inner values that are null to avoid edge cases that would cause the logic to fail (like if someone instantiates a type without using the factory) and ensure that the code paths are not activated when the value we are filtering on is "nullorempty".